lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -634,6 +640,38 @@ final class GenericRole extends PrimaryRoleManager {
       setAriaRole('text');
     }
   }
+
+  @override
+  bool focusAsRouteDefault() {
+    // Case 1: current node has input focus. Let the input focus system decide
+    // default focusability.
+    if (semanticsObject.isFocusable) {
+      final Focusable? focusable = this.focusable;
+      if (focusable != null) {
+        return focusable.focusAsRouteDefault();
```

Comment:
I think it would be confusing if a node is explicitly focusable but when its focus management logic rejects focus we look for other ways for it to take it. It's probably better to look for another node that's more willing to take focus. But maybe I'm missing something. Do you have an example where we'd want to insist that a node take focus even if its focus management doesn't want it?

Reply:
I see. Let's take a disabled button, for example. The button does not take input focus because it's disabled. However, a screen reader can still find it via traversal, so it could theoretically be as much of an auto-focus candidate as static text is. Is this what you mean?

I'll need to test what effect calling `.focus()` on disabled controls has. I imagine it may be a noop, or it may depend on whether an element carries intrinsic semantics/focusability (e.g. `<button>`, `<input>`), or whether it's just a "div" with ARIA attributes on it.

Reply:
`.focus()` is a noop on disabled native controls unless they have any valid tabindex on it (including -1), thereby allowing programmatic focus, as you're already aware below.
The disabled button is an interesting example. I think with that you've sold me.
But to confirm: are focusable entities that don't get primary focus considered focusable? If so, then I wonder whether one should recurse.

Reply:
Thanks. What I'll do is keep it as is for now, and wait for real world feedback to see how this logic could be improved. Even with the corner cases, I think the new approach already improves the situation considerably.